packages/@aws-cdk/core/README.md

```diff
@@ -94,8 +94,28 @@ nested stack and referenced using `Fn::GetAtt "Outputs.Xxx"` from the parent.
 
 Nested stacks also support the use of Docker image and file assets.
 
+## Weak References
 
-## Durations
+The CDK automatically wires up cross-stack references within a single app. By default, the CDK uses the CloudFormation's
+[`Fn::ImportValue`](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/intrinsic-function-reference-importvalue.html)
+instrinsic to achieve this. To prevent interruption, CloudFormation restricts modification of any `Output` of a stack
+that is imported via this intrinsic.
+
+In many cases, this can be too restrictive. Some resources guarantee that change to its physical name does not entirely
+delete the resource. Weak references relax this restriction for specific Constructs. Instead of using the
+`Fn::ImportValue` intrinsic, [SSM Parameter
+Types](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/parameters-section-structure.html#aws-ssm-parameter-types)
+are used instead.
+
+Enable weak reference for a resource:
+
+```ts
+resource.node.enableWeakReferences();
+```
+
+## CDK Types
```

Comment:
Why do we need this extra nesting?

Reply:
Makes it cleaner I hope.

Reply:
Please keep flat. Nesting usually makes things less readable in my experience.

Reply:
Sure!